hs_communities/static/js/community.js

```diff
@@ -0,0 +1,292 @@
+        joined: null,
+      },
+      cardPosition: {
+        top: 0,
+        left: 0,
+      }
+    },
+    computed: {
+      filteredAvailableToInvite() {
+        if (!this.inviteSearch) {
+          return this.availableToInvite
+        }
+        return this.availableToInvite.filter(group => {
+          return group.name.toLowerCase().indexOf(this.inviteSearch) >= 0
+        })
+      }
+    },
+    beforeMount() {
+      // Load data
+      const appData = JSON.parse(document.getElementById('community-app-data').textContent);
+
+      this.community = appData.community;
+      this.allCommunities = appData.all_communities;
+      this.isAdmin = appData.is_admin;
+      this.availableToInvite = appData.groups;
+      this.members = appData.members;
+      this.pending = appData.pending;
+    },
+    mounted() {
+      // Styling and placeholder for user auto-complete
+      $("input[name='user-autocomplete']")
+        .attr("placeholder", "Search by name or username")
+        .addClass("form-control");
+
+      // Initialize DataTables filter data
+      const groupIds = {};
+
+      $('#groups-list li').each(function () {
+        const groupId = parseInt($(this).attr('id'));
+        groupIds[$(this).text()] = groupId;
+      });
+
+      this.$data.groupIds = groupIds;
+
+      const filterGroup = $('#filter-querystring').text();
+      if (filterGroup && this.$data.groupIds[filterGroup]) {
+        this.$data.filterTo.push(this.$data.groupIds[filterGroup])
+      }
+    },
+    methods: {
+      loadOwnerCard(data) {
+        const el = $(data.event.target).closest('.profile-link');
+        const cardWidth = 350;
+
+        this.userCardSelected = data.user;
+        this.cardPosition.left = el.position().left - (cardWidth / 2) + (el.width() / 2);
+        this.cardPosition.top = el.position().top + 30;
+      },
+      isVisible(groupId) {
+        if (this.$data.filterTo.length === 0) {  // If no selections show all
+          return true;
+        } else {  // Display row if Group ID found in the filterTo Array
+          return this.$data.filterTo.indexOf(groupId) > -1;
+        }
+      },
+      updateContributors(groupId) {
+        const loc = this.$data.filterTo.indexOf(groupId);
+
+        if (loc < 0) {
+          this.$data.filterTo.push(groupId)
+        } else {
+          this.$data.filterTo.splice(loc, 1);
+        }
+      },
+      removeGroup: async function (id) {
+        this.$set(this.isRemoving, id, true)
+        const url = '/access/_internal/community/' + this.community.id + '/remove/' + id + '/';
+        $("#remove-group-modal").modal('hide')
+        try {
+          const response = await $.post(url)
+          this.members = response.members
+          this.availableToInvite = response.groups
+          customAlert("Remove Group", 'Group has been removed from your Community', "success", 6000, true);
+        }
+        catch (e) {
+          console.log(e)
+          // abort
+        }
```

Comment:
do we need to do anything else to handle this?

Reply:
Added message.